src/sizes/retained_size_by_objects.cpp

```diff
@@ -69,23 +69,23 @@ jvmtiError RetainedSizeByObjectsAction::retagStartObjects(const std::vector<jobj
 jvmtiError RetainedSizeByObjectsAction::tagHeap(const std::vector<jobject> &objects) {
     jvmtiError err = FollowReferences(0, nullptr, nullptr, getTagsWithNewInfo, nullptr, "find objects with new info");
     if (!isOk(err)) return err;
-    if (shouldStopExecution()) return MEMORY_AGENT_TIMEOUT_ERROR;
+    if (shouldStopAction()) return MEMORY_AGENT_INTERRUPTED_ERROR;
```

Comment:
Looks a bit strange that we use three different ways to check if the process is canceled:

```
// with passing arguments:
shouldStopExecution(finishTime, cancellationFileName)

// check using this:
this->shouldStopExecution()

// without any quilifier
shouldStopAction()
```

A good practice is to force using the only way to check the cancellation. Otherwise, we'll encounter more bugs and code becomes harder to understand and refactor.

Reply:
`shouldStopAction()` is a member funtion of MemoryAgentTimeoutBase. It is equal to `shouldStopExecution(finishTime, cancellationFileName)`, because it just passes action's parameters to this method. `shouldStopExecution(finishTime, cancellationFileName)` gives the ability to check interruption for static functions that are used by many actions. `this->shouldStopAction()` is the same that `shouldStopAction()` but we must write `this->` because we want to access `shouldStopAction()` in a template class that is inherited from base template class.